Code/Framework/AzToolsFramework/Tests/Prefab/Overrides/PrefabInspectorOverrideTests.cpp

```diff
@@ -31,24 +31,42 @@ namespace UnitTest
+            // Validate the translate row has overridden data.
+            // <Row>
+            //   <PropertyEditor Type="PrefabOverrideLabel" Text="Translate" IsOverridden="true" />
+            //   <PropertyEditor ValueType="Vector3" Value=... />
+            // </Row>
+            AZ::Dom::Path pathToTranslateRow(translateRowOrigin);
+            AZ::Dom::Value translateRow = adapterContents[pathToTranslateRow];
             EXPECT_EQ(translateRow.GetType(), AZ::Dom::Type::Node);
-            EXPECT_EQ(translateRow.ArraySize(), 3);
-
-            // First comes the PrefabOverrideProperty which will be used later by the PrefabAdapter to put up the PrefabOverrideIcon.
-            EXPECT_EQ(
-                translateRow[0][AZ::DocumentPropertyEditor::Nodes::PropertyEditor::Type.GetName()].GetString(),
-                AzToolsFramework::Prefab::PrefabPropertyEditorNodes::PrefabOverrideProperty::Name);
-
-            // Then comes the label of the row.
-            EXPECT_EQ(translateRow[1][AZ::DocumentPropertyEditor::Nodes::Label::Value.GetName()].GetString(), "Translate");
+            EXPECT_EQ(translateRow.ArraySize(), 2);
 
-            // Then comes the actual property editor of the translate vector.
-            AZ::Dom::Value translateVectorInDom = translateRow[2][AZ::DocumentPropertyEditor::Nodes::PropertyEditor::Value.GetName()];
-            EXPECT_EQ(translateVectorInDom[0].GetDouble(), 10.0);
+            AZ::Dom::Value labelPropertyEditor = translateRow[0];
+            EXPECT_EQ(labelPropertyEditor[PropertyEditor::Type.GetName()].GetString(), PrefabOverrideLabel::Name);
+            EXPECT_EQ(labelPropertyEditor[PrefabOverrideLabel::Text.GetName()].GetString(), "Translate");
+            EXPECT_TRUE(labelPropertyEditor[PrefabOverrideLabel::IsOverridden.GetName()].GetBool());
+
+            AZ::Dom::Value valuePropertyEditor = translateRow[1];
+            EXPECT_EQ(valuePropertyEditor[PropertyEditor::Value.GetName()][0].GetDouble(), 10.0);
+
+            // Validate the parent row has overridden data since child rows have overridden data.
+            // <Row>
+            //   <PropertyEditor Type="PrefabOverrideLabel" Text="Values" IsOverridden="true" />
+            //   <Row> ... </Row>  <-- translate row
+            // </Row>
+            AZ::Dom::Path pathToParentRow(parentRowOrigin);
+            AZ::Dom::Value parentRow = adapterContents[pathToParentRow];
+            EXPECT_EQ(parentRow.GetType(), AZ::Dom::Type::Node);
+
+            AZ::Dom::Value labelPropertyEditorInParentRow = adapterContents[pathToParentRow][0];
+            EXPECT_EQ(labelPropertyEditorInParentRow[PropertyEditor::Type.GetName()].GetString(), PrefabOverrideLabel::Name);
+            EXPECT_EQ(labelPropertyEditorInParentRow[PrefabOverrideLabel::Text.GetName()].GetString(), "Values");
+            EXPECT_TRUE(labelPropertyEditorInParentRow[PrefabOverrideLabel::IsOverridden.GetName()].GetBool());
```

Comment:
Are we sure we want this validation here? 

Long term, we want the icon on the parent row to show up only when the row data is collapsed right? So we need to remove this code later and add new tests for 'expand/collapse' behavior anyway

Reply:
Yeah I didn't recall the 'expand/collapse' case. Will remove it for now.